### PR TITLE
feat: Use URL escape for DTO name field

### LINKED ISSUE #850: feat: Use URL escape for DTO name field
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

 This feature request is for core-metadata
https://github.com/edgexfoundry/edgex-go/issues/4053

### Description [**REQUIRED**]
Use URL escape for the DTO name field,
- device name
- profile name
- service name
- interval and interval action name
- provision watcher name
- subscription name

The implementation can refer to  https://github.com/edgexfoundry/go-mod-core-contracts/pull/806.

### PR DESCRIPTION
**Pull request by weichou1229**
Use URL escape for DTO name field, including
- device name
- profile name
- service name
- interval and interval action name
- provision watcher name
- subscription name

Close #850

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [v] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [v] I am not introducing a new dependency (add notes below if you are)
- [v] I have added unit tests for the new feature or bug fix (if not, why?)
- [v] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) not impact
  <link to docs PR>

## Testing Instructions

Test core metadata API to verify the deviceName or serviceName accept URL escape.

### CODE REVIEW

**Comment by cloudxxx8** [comment]
do we still need `edgex-dto-rfc3986-unreserved-chars` validation fucntion?

**Comment by weichou1229** [comment]
There are DTOs still using the `edgex-dto-rfc3986-unreserved-chars`. 
https://github.com/edgexfoundry/go-mod-core-contracts/blob/main/dtos/subscription.go
https://github.com/edgexfoundry/go-mod-core-contracts/blob/main/dtos/notification.go
Maybe we can also remove these restrictions?

**Comment by cloudxxx8** [comment]
ok, let's keep it as is

File: dtos/requests/interval_test.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func TestAddIntervalRequest_Validate(t *testing.T) {
 		{"valid AddIntervalRequest", valid, false},
 		{"valid AddIntervalRequest, no Request Id", noReqId, false},
 		{"valid AddIntervalRequest, interval name containing unreserved chars", intervalNameWithUnreservedChars, false},
-		{"invalid AddIntervalRequest, interval name containing reserved chars", intervalNameWithReservedChars, true},
+		{"Valid AddIntervalRequest, interval name containing reserved chars", intervalNameWithReservedChars, false},
```
Comment (cloudxxx8) [inline dtos/requests/interval_test.go:90]: lowercase to meet the original test message

File: dtos/requests/intervalaction_test.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func TestAddIntervalActionRequest_Validate(t *testing.T) {
 		{"valid AddIntervalActionRequest", valid, false},
 		{"valid AddIntervalActionRequest, no Request Id", noReqId, false},
 		{"valid AddIntervalActionRequest, interval name containing unreserved chars", intervalNameWithUnreservedChars, false},
-		{"invalid AddIntervalActionRequest, interval name containing reserved chars", intervalNameWithReservedChars, true},
+		{"Valid AddIntervalActionRequest, interval name containing reserved chars", intervalNameWithReservedChars, false},
```
Comment (cloudxxx8) [inline dtos/requests/intervalaction_test.go:93]: lowercase to meet the original test message

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: dtos/requests/interval_test.go (hunk 1 above)
Comment (weichou1229) [reply dtos/requests/interval_test.go]: Updated.

File: dtos/requests/intervalaction_test.go (hunk 2 above)
Comment (weichou1229) [reply dtos/requests/intervalaction_test.go]: Updated.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**